### PR TITLE
feat(ui-icons): release new icons

### PR DESCRIPTION
**Pull request by ismay**
A new PR, based on https://github.com/dhis2/ui/pull/313. See https://jira.dhis2.org/browse/LIBS-86 for the related Jira issue. Note that this PR is a **BREAKING CHANGE**, so if merged this'll release a new major version (see 68efe2a).

I recommend using [this filter](https://github.com/dhis2/ui/pull/395/files?file-filters%5B%5D=.feature&file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.lock&file-filters%5B%5D=.md&file-filters%5B%5D=.snap&file-filters%5B%5D=dotfile) for reviewing, so that you won't have several hundred svgs in the diff.

- [x] Add svgs
- [x] Replace hardcoded colors with `inherit` keyword
- [x] Build needs to run before storybook generation (done by our netlify config, not dealt with for local builds)
- [x] Reference all issues and PRs that should be closed by this PR
- [x] Update local docs
- [x] Add tests to ensure the build is working as expected
- [x] Fix yarn installation issues
- [x] Ensure publishing settings are correct (main, module, etc.)

Notes:

* The named export's identifier does not match the component displayName. No way to adjust that with svgr-cli, if ever we decide we want to change that we should probably use svgr's node lib. (See: https://github.com/gregberge/svgr/issues/501)
* Our storybook setup (which used a private package) was causing problems with yarn installing dependencies (I assume due to the noHoist). I've tried resolving it with just noHoist, but the only reliable solution was refactoring it to be part of the root package. Functionality is still the same.
* I was adding build instructions and docs for ui-icons, and couldn't really do that without slightly rewriting the existing docs. Hopefully everyone is ok with the changes I've made, I feel they make the docs more user friendly, but let me know if you disagree.
* There were a couple of flaky cypress select tests that were giving me trouble. I've never been really happy with those tests so I refactored them to something that made a bit more sense to me.

Follow up:

* Create docs that allow users to easily find available icons. Currently we just link to the src folder.
* Move the removed parts of CONTRIBUTING.md to a notes discussion after merging this.

Closes #313, #284, #396, https://github.com/dhis2/notes/issues/175, https://github.com/dhis2/notes/issues/163, https://github.com/dhis2/notes/issues/162, https://github.com/dhis2/notes/issues/160, https://github.com/dhis2/notes/issues/159, https://github.com/dhis2/notes/issues/158

### CODE REVIEW

**Comment by ismay** [comment]
I suspect the e2e failures are because storybook isn't loading the correct font yet.

File: packages/icons/src/svg/filter-16.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg height="16" viewBox="0 0 16 16" width="16" xmlns="http://www.w3.org/2000/svg"><path d="m10 10-1 1h-2l-1-1zm3-3-1 1h-8l-1-1zm2-3-1 1h-12l-1-1z" fill="#404b5a" fill-rule="evenodd"/></svg>
```
Comment (cooper-joe) [inline packages/icons/src/svg/filter-16.svg:1]: The `fill` color is not set correctly, this is due to an error when exporting the source files. This icon will not build correctly if `fill: #010101` is not set, right?

File: packages/icons/src/svg/dimension-category-option-groupset-16.svg [hunk 2]
```diff
@@ -0,0 +1 @@
+<svg height="16" viewBox="0 0 16 16" width="16" xmlns="http://www.w3.org/2000/svg"><path d="m14 1c.5522847 0 1 .44771525 1 1v12c0 .5522847-.4477153 1-1 1h-12c-.55228475 0-1-.4477153-1-1v-12c0-.55228475.44771525-1 1-1zm0 1h-12v12h12zm-5 8-1 1-1-1zm2-2-1 1h-4l-1-1zm2-2-1 1h-8l-1-1z" fill="#404b5a" fill-rule="evenodd"/></svg>
```
Comment (cooper-joe) [inline packages/icons/src/svg/dimension-category-option-groupset-16.svg:1]: `fill` rule not correctly set to `#010101` in the export process. Apologies!

File: packages/icons/src/svg/filter-16.svg (hunk 1 above)
Comment (ismay) [reply packages/icons/src/svg/filter-16.svg:1]: No worries, good catch! Yeah it looks for `#010101` when doing the search and replace.

File: packages/icons/src/svg/dimension-category-option-groupset-16.svg (hunk 2 above)
Comment (ismay) [reply packages/icons/src/svg/dimension-category-option-groupset-16.svg:1]: Thanks for the heads up, I'll update it 👍.

Comment (ismay) [reply packages/icons/src/svg/dimension-category-option-groupset-16.svg]: Fixed in ab026e1

File: packages/icons/src/svg/filter-16.svg (hunk 1 above)
Comment (ismay) [reply packages/icons/src/svg/filter-16.svg]: Fixed in ab026e1

**Review by HendrikThePendric — APPROVED** [review]
@ismay and I discussed this PR on Slack. I requested a change to the `CONTRIBUTING.md` where it said `snake-case` which should have been `kebab-case`. And more importantly I requested to change the icon component template so that the component only accepts the `color` prop instead of spreading arbitrary props. Both of these points have been addressed, so I am approving.

**Comment by varl** [comment]
@ismay In the examples for how to import an icon it says to import from `@dhis2/ui`, but when I look at `ui/src/index.js` I don't see the icons re-exported: https://github.com/dhis2/ui/blob/update-ui-icons/packages/ui/src/index.js

Is that done in a build step or generated somewhere?

Thanks for the filter in the PR btw, very thoughtful. :+1:

**Comment by varl** [comment]
@ismay we discussed shipping the source SVG's in the package as well so it would be possible to do e.g. `import Icon from `@dhis2/ui-icons/svg/arrow-up-24.svg`.

Was this feature dropped or postponed to a later release? Right now only the `build` directory is in the `files` array of `package.json`: https://github.com/dhis2/ui/blob/update-ui-icons/packages/icons/package.json#L26

**Comment by ismay** [comment]
> @ismay we discussed shipping the source SVG's in the package as well so it would be possible to do e.g. `import Icon from '@dhis2/ui-icons/svg/arrow-up-24.svg'`.
> 
> Was this feature dropped or postponed to a later release? Right now only the `build` directory is in the `files` array of `package.json`: https://github.com/dhis2/ui/blob/update-ui-icons/packages/icons/package.json#L26

I understood that we wanted to postpone shipping the svg's for now. And that for the first release we wanted to focus on a minimal release, and then add further features later on.

**Comment by ismay** [comment]
> @ismay In the examples for how to import an icon it says to import from `@dhis2/ui`, but when I look at `ui/src/index.js` I don't see the icons re-exported: https://github.com/dhis2/ui/blob/update-ui-icons/packages/ui/src/index.js
> 
> Is that done in a build step or generated somewhere?
> 
> Thanks for the filter in the PR btw, very thoughtful. 👍

Ah that's a good catch, I forgot to look at that. Will add a reexport there.

**Comment by varl** [comment]
I cannot start Storybook locally, I get a lot of stack traces, e.g.:

```
[watch:storybook] ERROR in ./.storybook/preview.js-generated-config-entry.js
[watch:storybook] Module build failed (from ./node_modules/react-scripts/node_modules/babel-loader/lib/index.js):
[watch:storybook] Error: [BABEL] /home/varl/dev/dhis2/libs/ui/.storybook/preview.js-generated-config-entry.js: Cannot find module 'levenary'
[watch:storybook] Require stack:
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/babel-preset-react-app/node_modules/@babel/helper-compilation-targets/lib/index.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/babel-preset-react-app/node_modules/@babel/preset-env/lib/debug.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/babel-preset-react-app/node_modules/@babel/preset-env/lib/index.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/babel-preset-react-app/create.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/babel-preset-react-app/index.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/react-scripts/node_modules/@babel/core/lib/config/files/plugins.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/react-scripts/node_modules/@babel/core/lib/config/files/index.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/react-scripts/node_modules/@babel/core/lib/index.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/react-scripts/node_modules/babel-loader/lib/index.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/loader-runner/lib/loadLoader.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/loader-runner/lib/LoaderRunner.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/webpack/lib/NormalModule.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/webpack/lib/NormalModuleFactory.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/webpack/lib/Compiler.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/webpack/lib/webpack.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/@storybook/core/dist/server/build-static.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/@storybook/core/server.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/@storybook/react/dist/server/index.js
[watch:storybook] - /home/varl/dev/dhis2/libs/ui/node_modules/@storybook/react/bin/index.js (While processing: "/home/varl/dev/dhis2/libs/ui/node_modules/babel-preset-react-app/index.js")
[watch:storybook]     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
[watch:storybook]     at Function.Module._load (internal/modules/cjs/loader.js:725:27)
[watch:storybook]     at Module.require (internal/modules/cjs/loader.js:952:19)
[watch:storybook]     at require (internal/modules/cjs/helpers.js:88:18)
[watch:storybook]     at Object.<anonymous> (/home/varl/dev/dhis2/libs/ui/node_modules/babel-preset-react-app/node_modules/@babel/helper-compilation-targets/lib/index.js:41:40)
[watch:storybook]     at Module._compile (internal/modules/cjs/loader.js:1063:30)
[watch:storybook]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
[watch:storybook]     at Module.load (internal/modules/cjs/loader.js:928:32)
[watch:storybook]     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
[watch:storybook]     at Module.require (internal/modules/cjs/loader.js:952:19)
[watch:storybook]     at require (internal/modules/cjs/helpers.js:88:18)
[watch:storybook]     at Object.<anonymous> (/home/varl/dev/dhis2/libs/ui/node_modules/babel-preset-react-app/node_modules/@babel/preset-env/lib/debug.js:8:33)
[watch:storybook]     at Module._compile (internal/modules/cjs/loader.js:1063:30)
[watch:storybook]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
[watch:storybook]     at Module.load (internal/modules/cjs/loader.js:928:32)
[watch:storybook]     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
[watch:storybook]  @ multi ./node_modules/react-scripts/node_modules/@pmmmwh/react-refresh-webpack-plugin/client/ReactRefreshEntry.js ./node_modules/@storybook/core/dist/server/common/polyfills.js ./node_modules/@storybook/core/dist/server/preview/globals.js ./.storybook/storybook-init-framework-entry.js ./.storybook/preview.js-generated-config-entry.js ./.storybook/generated-stories-entry.js (webpack)-hot-middleware/client.js?reload=true&quiet=false&noInfo=true ./node_modules/react-scripts/node_modules/react-dev-utils/webpackHotDevClient.js main[4]
```

**Comment by ismay** [comment]
Yeah I suspect that remove all node_modules (also any that might be in ./packages or ./storybook) and then reinstalling will resolve that. Or maybe a rebuild.

**Comment by varl** [comment]
I'll do a full nuke then, I already tried rebuilding.

**Review by varl — APPROVED** [review]
Nice work! :tada:

**Review by cooper-joe — APPROVED** [review]
LGTM 👍

**Pull request merged**